libr/core/cmd_search.c

```diff
@@ -1677,14 +1656,14 @@ r_list_foreach (param->boundaries, iter, map) {
 						searchprefix, kwidx, count);
 					r_flag_set (core->flags, flag, at, ret);
 				}
-				if (*cmdhit) {
+				if (*param->cmd_hit) {
```

Comment:
same problem here

Reply:
I dislike these unnecessary checks. There should be a converse assert, which is defined as an empty in develop build (so we can learn what checks are unnecessary), and a `if (!xxxx) return NULL` in normal build / release build to satisfy nullness check fans.

Reply:
I dislike unexpected crashes just to save a simple check...

Reply:
agree with @rlaemmert

Reply:
Adding more nullchks makes the code more stable for corner use cases, and we want to not crash under any situation if we can manage it. so pls add it

Reply:
compilers can always remove code, but they will never add more code

Reply:
Added a stupid check when initializing param

```
	if (!param.cmd_hit) {
		param.cmd_hit = "";
	}
```

Reply:
@rlaemmert I do not lean too much towards "removing the null deref check" here. But I can not accept the stupid check to rbtree.c https://github.com/radare/radare2/commit/899d56157409e6273f8eed143123dbc84d50a4e2 at all.

Reply:
It really doesnt matter what you can accept. As long as nobody creates perfect code we need
to be aware of crashes. And the so called stupid simple check prevents us from that...
Easy as that. In the end of the day i want to use the tool, if the tool crashes unexpected my
whole non saved labels/comments and such may be lost.

Reply:
Checks are more demanded in API boundaries but for such deeply internal red-black tree detail. I hate the q->red && ***p*** && p->red so much. This is very intricate, I have been very carefully to use `x->red` or `red (x)` (with NULL deref check). The unnecessary check impairs the integrity, I would wonder when `p` would be NULL while actually it could not.

If I considered `r_config_get` as a distant API, I would accept the NULL deref check. But I cannot stand that one in rbtree.c